pkgs/top-level/all-packages.nix

```diff
@@ -6185,15 +6185,20 @@ in
     wxGTK = wxGTK30;
   };
 
-  buildbot = callPackage ../development/tools/build-managers/buildbot {
-    inherit (pythonPackages) twisted jinja2 sqlalchemy_migrate_0_7;
-    dateutil = pythonPackages.dateutil_1_5;
-  };
-
+  buildbot8 = callPackage ../development/tools/build-managers/buildbot/8.nix { };
   buildbot-slave = callPackage ../development/tools/build-managers/buildbot-slave {
     inherit (pythonPackages) twisted;
   };
 
+  buildbot = callPackage ../development/tools/build-managers/buildbot { };
+  buildbot-plugins = callPackage ../development/tools/build-managers/buildbot/plugins.nix { };
+  buildbot-ui = callPackage ../development/tools/build-managers/buildbot {
+    plugins = with self.buildbot-plugins; [ www ];
+  };
+  buildbot-full = callPackage ../development/tools/build-managers/buildbot {
+    plugins = with self.buildbot-plugins; [ www console-view waterfall-view ];
+  };
+
```

Comment:
Is it needed to rebuild `buildbot`, just to include plugins? Can you not build an environment that puts the basic `buildbot` and plugins together?

We have a `python.withPackages ()` which can be used to build an environment that includes certain Python packages, and it looks ideal for this use case.